docs/gateways/paypal.md

```diff
@@ -45,7 +45,49 @@ To redirect the customer off to PayPal's checkout page, you can use the `sc:chec
 
 However, bear in mind that where-ever you use that tag, the customer will be redirected away from your site. So it's probably best to have it sitting on it's own page.
 
-### Handling PayPal's webhook
+### Handling the redirect back to your site
+
+On the return back to your site from PayPal, you can have customers redirected to seperate URLs, depending on whether the payment was successful or failed/cancelled.
+
+The `redirect` parameter on the `sc:checkout:paypal` tag will handle the successful payment redirects.
+
+Where as `error_redirect` will handle any other payment states.
+
+## On-site payment flow
+
+Set the gateway configuretion to use on-site mode:
+
+```php
+'gateways' => [
+	\DoubleThreeDigital\SimpleCommerce\Gateways\Builtin\PayPalGateway::class => [
+        ///
+        'mode' => 'onsite',
+    ],
+],
+```
+
+The payment form should be included inside your `{{ sc:checkout }}` form, and any PayPal magic should also be wrapped in the `{{ sc:gateways }}` tag to ensure you can make full use of your gateway's configuration values.
+
+A rough example of a PayPal implementation is provided below.
+
+```antlers
+<div id="paypal-button"></div>
+<input id="paypal-payment-id" type="hidden" name="payment_id">
+<script src="https://www.paypal.com/sdk/js?client-id={{ gateway-config:client_id }}&currency={{ paypal.result.currency_code }}"></script>
+<script>
+    paypal.Buttons({
+        createOrder: () => {
+            return Promise.resolve('{{ paypal.result.id }}');
```

Comment:
`There seems to be a glitch where the first time the payment page is loaded the {{ paypal.result.* }} variables aren't defined. However if you refresh the page it works fine. Not sure if this is an SC glitch or I've set something up wrong in the starter.`

If you adjust this to just being `result.id`, it'll work first time. It's probably trying to use an old version of the entry (before the paypal data is saved). Whereas, the `result` is just passed in by the tag from the gateway prepare. At least I _think_ that's what's happening 😅

Reply:
Ah ok cool, I'll give that a go.